pkg/runtimeproxy/server/docker/handler.go

```diff
@@ -102,39 +102,40 @@ func (d *RuntimeManagerDockerServer) HandleCreateContainer(ctx context.Context,
 		}
 		hookReq = podInfo.GetPodSandboxHookRequest()
 	}
-
-	hookResp, err := d.dispatcher.Dispatch(ctx, runtimeHookPath, config.PreHook, hookReq)
-	if err != nil {
-		klog.Errorf("Failed to call hook server %v", err)
-		http.Error(wr, err.Error(), http.StatusInternalServerError)
-		return
-	}
-
 	cfgBody := types.ConfigWrapper{
 		Config:           ContainerConfig,
 		HostConfig:       hostConfig,
 		NetworkingConfig: networkingConfig,
 	}
 
-	if runtimeResourceType == resource_executor.RuntimePodResource && hookResp != nil {
-		resp := hookResp.(*v1alpha1.PodSandboxHookResponse)
-		if resp.Resources != nil {
-			cfgBody.HostConfig = UpdateHostConfigByResource(cfgBody.HostConfig, resp.Resources)
-			podInfo.Resources = resp.Resources
-		}
-		cfgBody.HostConfig.CgroupParent = generateExpectedCgroupParent(d.cgroupDriver, resp.CgroupParent)
-		podInfo.CgroupParent = resp.CgroupParent
-	} else if hookResp != nil {
-		resp := hookResp.(*v1alpha1.ContainerResourceHookResponse)
-		if resp.ContainerResources != nil {
-			cfgBody.HostConfig = UpdateHostConfigByResource(cfgBody.HostConfig, resp.ContainerResources)
-			containerInfo.ContainerResources = resp.ContainerResources
+	if !types.SkipRuntimeHook(podInfo.Labels) {
+		hookResp, err := d.dispatcher.Dispatch(ctx, runtimeHookPath, config.PreHook, hookReq)
```

Comment:
which not set config.NoneRuntimeHookPath to hint skipping, just like HandleUpdateContainer/HandleStopContainer part? which may lead to less changes.

Reply:
+1